src/ibeo_core.cpp

```diff
@@ -946,6 +992,66 @@ std::vector<Point3DL> ScanData2208::get_scan_points()
   return v;
 }
 
+ScanData2209::ScanData2209() :
+  IbeoTxMessage(true, false, false)
+{}
+
+void ScanData2209::parse(const std::vector<uint8_t>& in)
+{
+  ibeo_header.parse(in);
+
+  uint16_t hdr = IBEO_HEADER_SIZE;
+
+  scan_start_time = read_be<NTPTime>(in, hdr);
+  scan_end_time_offset = read_be<uint32_t>(in, hdr + 8);
+
+  uint32_t flags = read_be<uint32_t>(in, hdr + 12);
+  fused_scan = ((flags & 0x00000100) > 0);
+  mirror_side = ((flags & 0x00000200) > 0) ? REAR : FRONT;
+  coordinate_system = ((flags & 0x00000400) > 0) ? VEHICLE : SCANNER;
+
+  scan_number = read_be<uint16_t>(in, hdr + 16);
+  scan_points = read_be<uint32_t>(in, hdr + 18);
+  number_of_scanner_infos = read_be<uint8_t>(in, hdr + 22);
+
+  // printf("%d SCAN POINTS REPORTED\n", scan_points);
+
+  for (uint8_t i = 0; i < number_of_scanner_infos; i++)
+  {
+    ScannerInfo2209 new_scanner_info;
+    new_scanner_info.parse(in, hdr + 26 + (i * 148));
+    scanner_info_list.push_back(new_scanner_info);
+  }
+
+  for (uint16_t i = 0; i < scan_points; i++)
+  {
+    ScanPoint2209 new_scan_point;
+    new_scan_point.parse(in, hdr + 26 + (number_of_scanner_infos * 148) + (i * 28));
+    scan_point_list.push_back(new_scan_point);
+  }
+}
+
+std::vector<Point3DL> ScanData2209::get_scan_points()
+{
+  std::vector<Point3DL> v;
+
+  for (ScanPoint2209 sp : scan_point_list)
+  {
+    if (sp.echo == 0 && sp.layer < 4 && !sp.transparent && !sp.ground && !sp.dirt && !sp.precipitation)
```

Comment:
Just out of curiosity, why must the layer be less than 4? 
Maybe more of a question for @JWhitleyAStuff since this line is the same for other existing scan points.

Reply:
In this case, it could be up to 8 since, at least I'm pretty sure, the Fusion system supports the LUX 8L. Normally, though, it would just be the LUX or LUX HD which only have 4 scan layers.

Reply:
Makes me wonder if we should define some constants named "LUX_LAYERS", "LUX_8L_LAYERS", etc. Maybe another PR though?